crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -69,6 +72,104 @@ struct CheckpointStoreTables {
 
     /// Stores certified checkpoints
     certified_checkpoints: DBMap<CheckpointSequenceNumber, CertifiedCheckpointSummary>,
+    /// Map from checkpoint digest to certified checkpoint
+    checkpoint_by_digest: DBMap<CheckpointDigest, CertifiedCheckpointSummary>,
+
+    /// Watermarks used to determine the highest verified and fully synced checkpoints
+    watermarks: DBMap<CheckpointWatermark, (CheckpointSequenceNumber, CheckpointDigest)>,
+}
+
+impl CheckpointStoreTables {
```

Comment:
I would prefer all those methods to be implemented on the CheckpointService. Is there reason why this could not be done or you find it concerning?

The reason is that I think there will be places where we will want to use some synchronization in the CheckpointService (for example for on epoch boundaries) and having those methods on *Tables will make this hard to do

Reply:
I don't quite understand your comment on synchronization, the table itself handles that (the db is safe to use in multithreaded environments)

Reply:
One example where this can be used - we have `CheckpointService::subscribe_checkpoints`, and we will want to notify subscribers of this method when we receive checkpoint through checkpoint sync, this requires access to notifiers in the CheckpointService. [note that I am not asking you to do that in this PR, I am just given example of why I want this code in the service and not tables]

Reply:
@andll Ok this is done now. I've made the tables struct private and have accessor methods on the the Service itself.